store/src/java/com/zimbra/cs/service/account/EndSession.java

```diff
@@ -44,10 +54,36 @@ public Element handle(Element request, Map<String, Object> context)
         if (clearCookies || getAuthenticatedAccount(zsc).isForceClearCookies()) {
             context.put(SoapServlet.INVALIDATE_COOKIES, true);
             try {
-				zsc.getAuthToken().deRegister();
-			} catch (AuthTokenException e) {
-				throw ServiceException.FAILURE("Failed to de-register an auth token", e);
-			}
+                AuthToken at = zsc.getAuthToken();
+                if (at.isJWT()) {
+                    String jti = at.getId();
+                    if (jti != null) {
+                        ZimbraLog.account.debug("EndSession: jti: %s",jti);
+                        ZimbraJWT jwt = JWTCache.get(jti);
+                        if (jwt != null) {
+                            String salt = jwt.getSalt();
+                            ZimbraLog.account.debug("EndSession: found salt in cache for jti: %s",jti);
```

Comment:
Is this code to retrieve SALT from cookie repeated in SoapEngine, if yes then maybe refactor.

Reply:
done.